api/v1/services/AuthService.js

```diff
@@ -65,3 +68,48 @@ module.exports.verify = function(token) {
 			throw new errors.UnprocessableRequestError(message);
 		});
 };
+
+
+/**
+ * Generates a token and deletes all existing tokens
+ * with the same scope.
+ * @param {User} user The user object to create a reset token for.
+ * @param {String} scope The scope to create the token for.
+ * @return {Promise<Bool>} Returns a Promise that resolves to
+ *                         true on a successful token creation.
+ */
+module.exports.generateToken = function(user, scope){
+	var tokenVal = utils.crypto.generateResetToken();
+	var user_id = user.get('id');
+
+	return Token
+		.where({user_id: user_id, type: scope}).fetchAll()
+		.then(function(tokens) {
+			return tokens.invokeThen('destroy')
+				.then(function() {
+					var token = Token.forge({type: scope, value: tokenVal, 'user_id': user_id});
+					logger.info('Password recovery token generated:' + tokenVal);
```

Comment:
Don't log this! We know a token has been generated because it's in the datastore. There's no reason to put this sensitive information in the log file too.